themes/bootstrap3/templates/record/cart-buttons.phtml

```diff
@@ -2,11 +2,11 @@
 <?php if ($cart->isActive()): ?>
   <?php $cartId = $this->source . '|' . $this->id; ?>
   <span class="btn-bookbag-toggle" data-cart-id="<?=$this->escapeHtmlAttr($this->id)?>" data-cart-source="<?=$this->escapeHtmlAttr($this->source)?>">
-    <a href="#" class="cart-add result-link icon-link hidden<?php if (!$cart->contains($cartId)): ?> correct<?php endif ?>">
+    <a href="#" class="cart-add result-link icon-link hidden<?php if (!$cart->contains($cartId)): ?> correct<?php endif ?>" aria-describedby="<?=$cartId?>">
```

Comment:
We should `$this->escapeHtmlAttr($cartId)` to ensure valid HTML.

Also note that this template can be rendered as part of the record view as well as in the search results. In the context of the record view, there will be no element with the ID of `$cartId` on the page. Maybe something like [RecordDriver/DefaultRecord/core.phtml](https://github.com/vufind-org/vufind/blob/dev/themes/bootstrap3/templates/RecordDriver/DefaultRecord/core.phtml#L45) needs to be adjusted to include an ID to avoid inconsistencies.

Reply:
@RLangeUni do you know about that problem in our environment?

Reply:
Well, the examples I just checked have the ID of the record as `$cartId`, that must be always available in the record view. Maybe I do misunderstand something here.

Reply:
@RLangeUni do you know how we handle that? Do you have an idea how to solve this?

Reply:
Indeed in finc we are using an extra attribut in cart/contents. I added it in commit vufind-org@c42074dde3.

Reply:
As for now there is no unique-id on the detail record page and the aria-describedby within the cart button is useless, but it should do no harm? Eventually we must avoid duplicate ids (I think the case should not occur).

Reply:
Maybe for this PR we can just move the `aria-describedby="<?=$cartId?>"` inside the if-condition `<?php if (!$cart->contains($cartId)): ?> correct<?php endif ?>" ` and make further improvements (like getUniqueIdWithSource()) in another PR? 
What do you think @demiankatz @RLangeUni ?

Reply:
I don't think that helps -- the issue is that aria-describedby needs to refer to an HTML element ID, but `$cartId` is an identifier used by the cart's Javascript code. There is no guarantee that a matching HTML element exists anywhere, so in many cases, this is going to be a broken reference that doesn't accomplish anything.